packages/framework/presence/src/presenceDatastoreManager.ts

```diff
@@ -195,14 +198,61 @@ export class PresenceDatastoreManagerImpl implements PresenceDatastoreManager {
 		return entry.public;
 	}
 
-	private localUpdate(data: DatastoreMessageContent, _forceBroadcast: boolean): void {
-		const content = {
-			sendTimestamp: Date.now(),
+	private queuedMessage: DatastoreUpdateMessage["content"] | undefined;
+
+	private sendMessageDeadline: number = 0;
+
+	// private timeoutId: number | undefined;
+
+	private localUpdate(data: DatastoreMessageContent, options: LocalUpdateOptions): void {
+		const allowableUpdateLatency = options.allowableUpdateLatency ?? 0;
+
+		const now = Date.now();
+		const updateDeadline = now + allowableUpdateLatency;
+		if (this.queuedMessage === undefined) {
+			// no queued message, so set the deadline based on the current message's allowable latency
+			this.sendMessageDeadline = updateDeadline;
+		}
+
+		const currentMessageData = data;
+		const queuedMessageData = this.queuedMessage?.data;
+
+		// Merge the queued data with the next update.
+		// TODO: this is not working as I expect. Data from multiple LVMs is not combined.
+		const newData: DatastoreMessageContent = { ...queuedMessageData, ...currentMessageData };
+
+		const newContent = {
+			sendTimestamp: now,
 			avgLatency: this.averageLatency,
 			// isComplete: false,
-			data,
+			data: newData,
 		} satisfies DatastoreUpdateMessage["content"];
-		this.runtime.submitSignal(datastoreUpdateMessageType, content);
+
+		if (updateDeadline >= this.sendMessageDeadline) {
+			// Queue the update
+			this.queuedMessage = newContent;
+			// if the timer has not expired, we can short-circuit because the timer will fire
+			// and cover this update
```

Comment:
```suggestion
			// and cover this update. in other words, queuing this will be fast enough to
			// meet its deadline, because a timer is already scheduled to fire before its deadline.
```